### PR TITLE
Reenable SonarCloud analysis on all PRs

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR re-enables SonarCloud analysis on all pull requests. I updated the organization since we've now changed it, and also changed it so that develop and master both act as "base" branches, i.e. they aren't compared against a different branch.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: .ci/scripts/distribution/analyse-java.sh [hunk 1]
```diff
@@ -23,12 +23,12 @@ else
...
 
   if [ "${GIT_BRANCH}" == "master" ] || [ "${GIT_BRANCH}" == "develop" ]; then
-    TARGET_BRANCH="master"
+    TARGET_BRANCH="${GIT_BRANCH}"
   else
     TARGET_BRANCH="develop"
+    PROPERTIES+=("-Dsonar.branch.target=${TARGET_BRANCH}")
   fi
 
-  PROPERTIES+=("-Dsonar.branch.target=${TARGET_BRANCH}")
```
Comment (korthout) [inline .ci/scripts/distribution/analyse-java.sh:31]: Can you explain what this fixes? It's unclear to me

Comment (npepinpe) [reply .ci/scripts/distribution/analyse-java.sh:31]: This doesn't fix anything, it was just unnecessary unless you were comparing for a PR.

**Review by korthout — APPROVED** [review]
LGTM ✔️

**Comment by npepinpe** [comment]
Ah, I think I see - the commit which said "fix integration" referred to changing the organization - if I kept it, it gave an error saying it was not the expected orga.

**Comment by npepinpe** [comment]
bors merge

**Comment by korthout** [comment]
> Ah, I think I see - the commit which said "fix integration" referred to changing the organization - if I kept it, it gave an error saying it was not the expected orga.

Yes, it was not clear what you fixed with that commit from the commit message.

**Pull request merged**